src/auxinfo/participant.rs

```diff
@@ -394,12 +394,20 @@ impl AuxInfoParticipant {
         }
 
         let decom = self.local_storage.retrieve::<storage::Decommit>(self.id)?;
-        let more_messages = self.message_for_other_participants(
-            MessageType::Auxinfo(AuxinfoMessageType::R2Decommit),
-            sid,
-            decom,
-        )?;
-        messages.extend(more_messages);
+        messages.extend(
+            self.other_participant_ids
+                .iter()
+                .map(|&other_participant_id| {
+                    Message::new(
+                        MessageType::Auxinfo(AuxinfoMessageType::R2Decommit),
+                        sid,
+                        self.id,
+                        other_participant_id,
+                        &decom,
+                    )
+                })
+                .collect::<Result<Vec<Message>>>()?,
+        );
```

Comment:
todo: Can you switch this back to using the `message_for_other_participants` method?